intellij/src/saros/intellij/editor/annotations/AbstractEditorAnnotation.java

```diff
@@ -205,9 +206,34 @@ void removeLocalRepresentation() {
    */
   void updateBoundaries() {
     annotationRanges.removeIf(
-        annotationRange ->
-            annotationRange.getRangeHighlighter() != null
-                && !annotationRange.getRangeHighlighter().isValid());
+        annotationRange -> {
+          RangeHighlighter rangeHighlighter = annotationRange.getRangeHighlighter();
+
+          if (rangeHighlighter == null) {
+            return false;
+          }
+
+          if (!rangeHighlighter.isValid()) {
+            return true;
+          }
+
+          if (rangeHighlighter.getStartOffset() == rangeHighlighter.getEndOffset()) {
+            if (editor != null) {
+              removeRangeHighlighter(editor, rangeHighlighter);
```

Comment:
Maybe a comment why you use `removeIf` even the lambda has side-effects. I guess you want the O(n) performance instead of O(n²).